src/pages/incomplete.js

```diff
@@ -1,35 +1,40 @@
 import React, { useEffect, useState } from "react"
+import { Typography } from "@material-ui/core"
 import { useDispatch, useSelector } from "react-redux"
-import { Box, Button, Typography } from "@material-ui/core"
 import { navigate } from "gatsby"
 import { selectAllEntries, upsertEntry } from "../features/entries"
+import { addMensesTag } from "../features/settings"
 import { useSettings } from "../features/settings"
-import LastDateInput from "../features/onboarding/Onboarding/LastDateInput"
 import { AppLayout, AppMainToolbar, AppPage } from "../features/app"
 import Toast from "../features/app/Toast"
 import { TIMELINE } from "../features/navigation"
+import { useSubscription } from "../features/user"
+import NoPayment from "../features/profile/Incomplete/NoPayment"
+import IncompleteSettings from "../features/profile/Incomplete/IncompleteSettings"
 
 const Incomplete = () => {
   const dispatch = useDispatch()
-
-  const [lastPeriod, setLastPeriod] = useState(new Date())
   const [error, setError] = useState(false)
 
   const { mainMensesTag } = useSettings()
 
   const entries = useSelector(selectAllEntries)
-  const notHasPlacedPeriod =
-    !entries.length || !entries.filter((s) => s.tags.length).length
+  const notHasPlacedPeriod = !entries.length
+  const { isSubscribed } = useSubscription()
 
   useEffect(() => {
     if (!notHasPlacedPeriod) {
       navigate(TIMELINE.to)
     }
   })
 
-  const calculatePeriodDate = async () => {
+  const onSubmit = async ({ tag, lastPeriod }) => {
     setError(false)
 
+    if (!mainMensesTag) {
+      await dispatch(addMensesTag(tag))
```

Comment:
**issue:** No dispatch on the functions from `useSettings`

Reply:
Yeah, already fixed but then Eurovision started 😂

Reply:
I approve of your priorities 🤣

Reply:
Euvosion is done and it is fixed :)